tests/test_cameraGeom.py

```diff
@@ -0,0 +1,108 @@
+        detectors from the same visit: they should have near identical pixel
+        scale variation.
+        """
+        # don't need the wcs or detector lists for this test
+        model = lsst.jointcal.cameraGeometry.CameraModel([], [], self.camera)
+        angle1, radial1, tangential1 = model._computeDetectorPixelScale(self.detector1,
+                                                                        self.wcs1)
+        angle2, radial2, tangential2 = model._computeDetectorPixelScale(self.detector2,
+                                                                        self.wcs2)
+        # The angles really should be identical: this value is independent of distortion.
+        self.assertFloatsEqual(angle1, angle2)
+        # These should be identical, but we allow for machine-precision errors
+        # due to possible complexity in the distortion model.
+        self.assertFloatsAlmostEqual(radial1, radial2)
+        self.assertFloatsAlmostEqual(tangential1, tangential2)
+        # Check that the HSC pixel scale is approximately correct at the boresight.
+        self.assertFloatsAlmostEqual(radial1[0], 0.168, rtol=1e-2)
+        self.assertFloatsAlmostEqual(tangential1[0], 0.168, rtol=1e-2)
+
+    def testComputeCameraPixelScale(self):
+        """Test CameraModel.computeCameraPixelScale by comparing two
+        detectors: they should have near identical pixel scale variation.
+        """
+        # don't need the wcs or detector lists for this test
+        model = lsst.jointcal.cameraGeometry.CameraModel([], [], self.camera)
+        angle1, radial1, tangential1 = model.computeCameraPixelScale(self.detector1)
+        angle2, radial2, tangential2 = model.computeCameraPixelScale(self.detector2)
+        # For the camera calculation, there can be machine-precision
+        # differences between different detectors, so "almost" here.
+        self.assertFloatsAlmostEqual(angle1, angle2)
+        self.assertFloatsAlmostEqual(radial1, radial2)
+        self.assertFloatsAlmostEqual(tangential1, tangential2)
+        # Check that the HSC pixel scale is approximately correct at the boresight.
+        self.assertFloatsAlmostEqual(radial1[0], 0.168, rtol=1e-3)
+        self.assertFloatsAlmostEqual(tangential1[0], 0.168, rtol=1e-3)
+
+    def testComputePixelScale(self):
+        """Test that the pixel scale calculation produces something sensible.
+        """
+        model = lsst.jointcal.cameraGeometry.CameraModel(self.wcsList, self.detectors, self.camera)
+        model.computePixelScale()
+        angle1, radial1, tangential1 = model._computeDetectorPixelScale(self.detector1,
+                                                                        self.wcs1)
+        # TOOD: If we ever go with a more careful approach to the mean
+        # fieldAngle question, this may have to change.
+        self.assertFloatsAlmostEqual(model.fieldAngle, angle1, rtol=1e-4)
```

Comment:
Unfortunately, I'm less sure how to test those: the fieldAngles should be very nearly identical between detectors on the same visit, but are slightly different across visits, whereas the other two are more different; that's the point of averaging across visits. Other than just repeating the math for the calculation here (which isn't really what I want) or putting in a similar "almost equal" test, I'm not sure what to do. Another "almost equal" test seems like it's too far from testing the important thing, which is the averaging; it would help check that the right axis was averaged over.

Reply:
Yeah, I see what you mean; most of the things I can think of are also repeating math or trivial.  That said, I don't think there's any harm in verifying that `model.fieldAngles[0] == angle1` and similarly `radial1` and `tangential1` (if I understand that correctly).